src/core/mod.rs

```diff
@@ -604,6 +613,134 @@ impl StacksEpochExtension for StacksEpoch {
         ]
     }
 
+    #[cfg(test)]
+    fn unit_test_2_2(first_burnchain_height: u64) -> Vec<StacksEpoch> {
```

Comment:
I guess we should complement these unit tests with integration tests similar to `testnet/stacks-node/src/tests/epoch_21.rs`. The 2.1 test file is huge though, so I don't know if it's possible to start small or if it's better to leave the full testing as a follow-up.

Reply:
You're right. I guess this should be included, although I wish it wasn't in this PR 😅. I'll take a look through that `epoch_21.rs` and start a version for 2.2.

Reply:
After looking at this, there's not really much interesting to test yet for 2.2, so I added a new file, but it just includes the `transition_empty_blocks` test for now.